src/components/HeaderBlock/HeaderBlock.js

```diff
@@ -0,0 +1,343 @@
+
+const HeaderTitle = styled(BaseText)`
+  line-height: ${fontSizes.small};
+  font-size: ${fontSizes.extraSmall}px;
+  color: ${props => props.theme.color || UIColors.defaultTextColor};
+  font-weight: 500;
+`;
+
+const UserButton = styled.TouchableOpacity`
+  flex-direction: row;
+  align-items: center;
+  margin-right: ${spacing.medium}px;
+`;
+
+const RightSide = styled.View`
+  flex-direction: row;
+  justify-content: flex-end;
+  flex: ${props => props.sideFlex ? props.sideFlex : 1};
+`;
+
+const LeftSide = styled.View`
+  flex-direction: row;
+  flex: ${props => props.sideFlex ? props.sideFlex : 1};
+  align-items: center;
+`;
+
+const MiddlePart = styled.View`
+  flex-direction: row;
+  flex-grow: 1;
+  align-items: center;
+  justify-content: center;
+`;
+
+const BackIcon = styled(IconButton)`
+  position: relative;
+  align-self: flex-start;
+  height: 44px;
+  width: 44px;
+  padding-left: 10px;
+  margin-left: -12px;
+  margin-bottom: -12px;
+`;
+
+const ActionIcon = styled(IconButton)`
+  position: relative;
+  align-self: center;
+  height: 24px;
+  width: 24px;
+  margin: 0 3px -6px;
+`;
+
+const TextButton = styled.TouchableOpacity`
+  padding: 5px;
+`;
+
+const ButtonLabel = styled(BaseText)`
+  line-height: ${fontSizes.small}px;
+  font-size: ${fontSizes.extraSmall}px;
+  color: ${props => props.theme.rightActionLabelColor || baseColors.electricBlue};
+`;
+
+const profileImageWidth = 24;
+
+const themes = (backgroundColor?: string = '') => ({
+  transparent: {
+    backgroundColor: 'transparent',
+    color: baseColors.slateBlack,
+    borderBottomColor: 'transparent',
+    borderBottomWidth: 0,
+    iconColor: baseColors.slateBlack,
+    rightActionIconColor: baseColors.electricBlue,
+    rightActionLabelColor: baseColors.electricBlue,
+    buttonBorderColor: baseColors.mediumLightGray,
+    buttonLabelColor: baseColors.coolGrey,
+  },
+  light: {
+    color: baseColors.white,
+    iconColor: baseColors.white,
+    rightActionIconColor: baseColors.white,
+    rightActionLabelColor: baseColors.white,
+    buttonBorderColor: UIColors.actionButtonBorderColor,
+    buttonLabelColor: baseColors.white,
+  },
+  background: {
+    backgroundColor,
+    borderBottomColor: backgroundColor,
+  },
+  darkBorder: {
+    borderBottomWidth: 1,
+    borderBottomColor: baseColors.slateBlack,
+  },
+  lighterHeader: {
+    color: baseColors.aluminium,
+  },
+  default: {
+    backgroundColor: baseColors.white,
+    color: baseColors.slateBlack,
+    borderBottomColor: baseColors.mediumLightGray,
+    borderBottomWidth: 1,
+    iconColor: baseColors.slateBlack,
+    rightActionIconColor: baseColors.electricBlue,
+    rightActionLabelColor: baseColors.electricBlue,
+    buttonBorderColor: baseColors.mediumLightGray,
+    buttonLabelColor: baseColors.slateBlack,
+  },
+});
+
+const getTheme = (props: Props) => {
+  const { background = '' } = props;
+  const combinedThemes = [];
+
+  Object.keys(props).forEach((prop: string) => {
```

Comment:
Oh, yeah - you are right :)
I'll update this part to not take into consideration any props that are set to false.
Thanks! 

And to answer your first question: In case of couple of props present - matching theme objects would be merged into one (overwriting values based on order) (this provides more flexible styles with less predefined style objects)

Reply:
[Here's the fix](https://github.com/pillarwallet/pillarwallet/pull/1285/commits/88c66429a5304c2dc574d59df493bddcbef76509#diff-54886ecd2f03ad48ab5df045bcbd2b11R197) :)